tests/src/test/330/scala/com/nvidia/spark/rapids/CsvScanForIntervalSuite.scala

```diff
@@ -0,0 +1,637 @@
+      s"interval '${Long.MaxValue / 1000000L + 1}' second"
+    )) { intervalCV =>
+      withResource(GpuIntervalUtils.castStringToDTInterval(intervalCV,
+        DayTimeIntervalType(DayTimeIntervalType.SECOND, DayTimeIntervalType.SECOND))) { actualCV =>
+        for (i <- 0 until actualCV.getRowCount.toInt)
+          assert(actualCV.isNull.getScalarElement(i).getBoolean) // return null because invalid
+      }
+    }
+
+    // check max leading second
+    withResource(ColumnVector.fromStrings(
+      s"interval '-${Long.MinValue / 1000000L + 1}' second"
+    )) { intervalCV =>
+      withResource(GpuIntervalUtils.castStringToDTInterval(intervalCV,
+        DayTimeIntervalType(DayTimeIntervalType.SECOND, DayTimeIntervalType.SECOND))) { actualCV =>
+        for (i <- 0 until actualCV.getRowCount.toInt)
+          assert(actualCV.isNull.getScalarElement(i).getBoolean) // return null because invalid
+      }
+    }
+  }
+
+  test("test castStringToDTInterval format invalid") {
+    withResource(ColumnVector.fromStrings(
+      "INTERVAL xx DAY",
+      "INTERVAL 3 day",
+      "INTERVAL 3' day",
+      "INTERVAL '3 day",
+      "INTERVAL '-      3' day",
+      "INTERVAL -       '3' day",
+      "INTERVAL '100 10:30:40.' DAY TO SECOND"
+    )) { intervalCV =>
+      withResource(GpuIntervalUtils.castStringToDTInterval(intervalCV,
+        DayTimeIntervalType(DayTimeIntervalType.DAY, DayTimeIntervalType.DAY))) { actualCV =>
+        for (i <- 0 until actualCV.getRowCount.toInt)
+          assert(actualCV.isNull.getScalarElement(i).getBoolean)
+      }
+    }
+  }
+
+  def readCsv(spark: SparkSession, path: String): DataFrame = {
+    def dayTime(s: Byte, e: Byte): DayTimeIntervalType = DayTimeIntervalType(s, e)
+
+    val schema = StructType(Seq(
+      StructField("c01", dayTime(DayTimeIntervalType.DAY, DayTimeIntervalType.DAY)),
+      StructField("c02", dayTime(DayTimeIntervalType.DAY, DayTimeIntervalType.HOUR)),
+      StructField("c03", dayTime(DayTimeIntervalType.DAY, DayTimeIntervalType.MINUTE)),
+      StructField("c04", dayTime(DayTimeIntervalType.DAY, DayTimeIntervalType.SECOND)),
+      StructField("c05", dayTime(DayTimeIntervalType.HOUR, DayTimeIntervalType.HOUR)),
+      StructField("c06", dayTime(DayTimeIntervalType.HOUR, DayTimeIntervalType.MINUTE)),
+      StructField("c07", dayTime(DayTimeIntervalType.HOUR, DayTimeIntervalType.SECOND)),
+      StructField("c08", dayTime(DayTimeIntervalType.MINUTE, DayTimeIntervalType.MINUTE)),
+      StructField("c09", dayTime(DayTimeIntervalType.MINUTE, DayTimeIntervalType.SECOND)),
+      StructField("c10", dayTime(DayTimeIntervalType.SECOND, DayTimeIntervalType.SECOND))
+    ))
+    fromCsvDf(path, schema)(spark)
+  }
+
+  testSparkResultsAreEqual(
+    "test read day-time interval csv file",
+    spark => readCsv(spark, "day-time-interval.csv")
+  ) {
+    df => df
+  }
+
+  /**
+   * TODO: Blocked by Spark overflow issue: https://issues.apache.org/jira/browse/SPARK-38520
```

Comment:
Do we do the right thing with the overflow? If so then we need to document this.

Reply:
Yes, should update doc, another followup PR: https://github.com/NVIDIA/spark-rapids/pull/4981